src/particles/diagnostics/ReducedBeamCharacteristics.cpp

```diff
@@ -283,6 +305,62 @@ namespace impactx::diagnostics
         amrex::ParticleReal const alpha_y = - ypy_d / emittance_yd;
         amrex::ParticleReal const alpha_t = - tpt / emittance_t;
 
+        // Calculate eigenemittances (optional)
+        amrex::Array2D<amrex::ParticleReal, 1, 6, 1, 6> Sigma;
+        amrex::ParticleReal emittance_1 = emittance_x * bg;
+        amrex::ParticleReal emittance_2 = emittance_y * bg;
+        amrex::ParticleReal emittance_3 = emittance_t * bg;
+
+        // Parse the diagnostic parameters
+        amrex::ParmParse pp_diag("diag");
+        bool compute_eigenemittances = false;
```

Comment:
My feeling is we should enable this by default, but maybe there is a good reason not to.
```suggestion
        bool compute_eigenemittances = true;
```

Reply:
The eigenemittance calculation is delicate for highly symmetric beams in uncoupled optics (where it is also not needed), and it causes one of the tests in pyTest to fail.  This, combined with its high complexity compared to other reduced diagnostics, suggests to me that it should be enabled if needed by the user.